applications/asset_tracker_v2/src/modules/data_module.c

```diff
@@ -1028,8 +1047,7 @@ static void agps_request_handle(struct nrf_modem_gnss_agps_data_frame *incoming_
 	 * A-GPS request and send out an event containing the request for the cloud module to pick
 	 * up and send to the cloud that is currently used.
 	 */
-	err = (incoming_request == NULL) ? agps_request_encode(NULL) :
-					   agps_request_encode(&request);
+	err = agps_request_encode(&request);
```

Comment:
Right now, the passed in `request` will never be NULL so the check `if (incoming_request == NULL)` inside `agps_request_encode` will never trigger.

Reply:
It's not necessary for `agps_request_encode()` to be passed NULL, because the code above in `agps_request_handle()` generates the correct request based on whether P-GPS is enabled or not.  I could move this code into `agps_request_encode()` instead.